spring-boot-autoconfigure/src/main/java/org/axonframework/springboot/autoconfig/MicrometerMetricsAutoConfiguration.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2010-2018. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.springboot.autoconfig;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.simple.SimpleMeterRegistry;
+import org.axonframework.micrometer.GlobalMetricRegistry;
+import org.axonframework.micrometer.MetricsConfigurerModule;
+import org.axonframework.springboot.MetricsProperties;
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.boot.autoconfigure.AutoConfigureBefore;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * Auto configuration to set up Micrometer Metrics for the infrastructure components.
+ *
+ * @author Steven van Beelen
+ * @author Marijn van Zelst
+ * @since 4.0
+ */
+@Configuration
+@AutoConfigureAfter(name = {
+        "org.springframework.boot.actuate.autoconfigure.metrics.CompositeMeterRegistryAutoConfiguration",
+        "org.springframework.boot.actuate.autoconfigure.metrics.export.simple.SimpleMetricsExportAutoConfiguration"
+})
+@AutoConfigureBefore(AxonAutoConfiguration.class)
+@ConditionalOnClass(name = {
+        "io.micrometer.core.instrument.MeterRegistry",
+        "org.axonframework.micrometer.GlobalMetricRegistry"
+})
+@EnableConfigurationProperties(MetricsProperties.class)
+public class MicrometerMetricsAutoConfiguration {
```

Comment:
Would you mind specifying that the old `MetricsAutoConfiguration` should happen after this guy? So with the `@AutoConfigureAfter()` annotation. I feel we should give precedence over the new approach set by Spring Boot in favor of the old one in the auto configuration set up.

Reply:
Ok I've done that. In the current setup both the old dropwizard metrics and the new micrometer metrics will be loaded simultaneously. Is that desirable?

Reply:
:thinking: you're specifying it _after_ the other one, so if anything, it's in order, not simultaneous, right?

Nonetheless, I'd expect that existence of the micrometer beans to ensure the Dropwizard-Metrics beans to _not_ be created. This holds as there's a `@ConditionalOnMissingBean` on the bean creator functions, as far as I am aware.

If my assumptions are incorrect here, please call me out on them @mvanzelst!

Reply:
It's a tricky one :smile: I've modified the autoconfig class of the metrics module to NOT load if the metrics module is loaded. Also I've added some tests to verify that if the metrics module beans are NOT loaded when the micrometer module is loaded. Would this be the best solution?

Reply:
I think we should take preference over Micrometer i.o. Dropwizard Metrics, so I'd say yes, this is the best solution. That's however my take on it. Maybe @abuijze / @m1l4n54v1c / @idugalic  have a different perspective on this.

Reply:
Yes, I agree